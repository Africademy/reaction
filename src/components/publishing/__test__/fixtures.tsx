import { Props as ImageSetPreviewProps } from "../imageset_preview"

export const Artworks = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
    },
    artist: {
      name: "Fernando Botero",
    },
    width: 1152,
    height: 826,
  },
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
      {
        name: "Frida Kahlo",
        slug: "frida-kahlo",
      },
    ],
    width: 1152,
    height: 826,
  },
]

export const Articles = [
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "classic",
    vertical: null,
  },
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "standard",
    vertical: {
      name: "Art Market",
      id: "12345",
    },
  },
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "feature",
    vertical: {
      name: "Art Market",
      id: "12345",
    },
  },
]

export const Authors = [
  {
    image_url: "https://artsy-media-uploads.s3.amazonaws.com/9vcX6FR21rKHatmvJ8K0sg%2FAbigail.jpg",
    name: "Abigail Cain",
    bio: "[Abigail Cain](https://artsy.net/abigail) is an Art Market Editor at Artsy",
    twitter_handle: "abigailcain",
  },
  {
    image_url: "http://files.artsy.net/images/molly_01.jpg",
    name: "Molly Gottschalk",
    bio:
      "[Molly Gottschalk](https://artsy.net/molly) is a Features Producer at Artsy. She is a photographer and lover of cats.",
    twitter_handle: "mollygottschalk",
  },
  {
    name: "Halley Johnson",
    bio: "[Halley Johnson](https://artsy.net/molly) is an Writer Guru at Artsy. She enjoys speaking Georgian.",
    twitter_handle: "halleyjohnson",
  },
]

export const Embeds = [
  {
    mobile_height: 1300,
    height: 1000,
    url: "http://files.artsy.net/documents/1parrasch.html",
    layout: "overflow",
    type: "embed",
  },
]

export const HeroSections = [
  {
    type: "text",
    url: "https://artsy-media-uploads.s3.amazonaws.com/YqTtwB7AWqKD95NGItwjJg%2FRachel_Rossin_portrait_2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "split",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "fullscreen",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "split",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "fullscreen",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "text",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
]

export const Images = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
    ],
    artist: {
      name: "Fernando Botero",
      slug: "fernando-botero",
    },
    width: 1152,
    height: 826,
  },
  {
    url: "https://artsy-media-uploads.s3.amazonaws.com/co8j2xq40ROMyBrJQm_4eQ%2FDafenOilPaintingVillage_AK03.jpg",
    type: "image",
    width: 900,
    height: 1200,
    caption: "<p>Photo by Adam Kuehl for Artsy.</p>",
  },
  {
    url: "https://d32dm0rphc51dk.cloudfront.net/CpHY-DRr7KW0HGXLslCXHw/larger.jpg",
    type: "image",
    width: 816,
    height: 1024,
    caption: "<p>Photo by Adam Kuehl for Artsy. Image courtesy of the Guggenheim Museum.</p>",
  },
]

export const ImageSetFull: ImageSetPreviewProps["section"] = {
  type: "image_set",
  layout: "full",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetMini: ImageSetPreviewProps["section"] = {
  type: "image_set",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetFullSansTitle: ImageSetPreviewProps["section"] = {
  type: "image_set",
  layout: "full",
  images: Images,
}

export const ImageSetMiniSansTitle: ImageSetPreviewProps["section"] = {
  type: "image_set",
  images: Images,
}
export const Videos = [
  {
    url: "https://www.youtube.com/watch?v=PXi7Kjlsz9A",
    caption: "<p>What motivates patrons to fund artists’ wildest dreams?</p>",
    cover_image_url: "https://artsy-media-uploads.s3.amazonaws.com/IB6epb5L_l0rm9btaDsY7Q%2F14183_MDP_Evening_240.jpg",
  },
  {
    url: "https://vimeo.com/191988155",
    caption:
      "<p>2016 was a memorable year for the world, and art along with it. Powered by data culled from Artsy as well as UBS’s Planet Art app, “The Year in Art 2016” will explore how the creative community responded to the cultural shifts and tribulations this year has seen—from the destruction of Palmyra to the proliferation of Virtual Reality to the U.S. election.</p>",
    cover_image_url: "https://artsy-media-uploads.s3.amazonaws.com/ditbyaUgdcl6mHin07TfKA%2FMassimilianoGioni_0581.jpg",
  },
  {
    url: "https://vimeo.com/191988155",
  },
]
