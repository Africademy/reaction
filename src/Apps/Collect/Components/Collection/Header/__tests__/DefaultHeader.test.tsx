import { defaultCollectionHeaderArtworks } from "Apps/Collect/Components/Collection/Header/__tests__/fixtures/artworks"
import {
  CollectionDefaultHeader,
  getHeaderArtworks,
} from "Apps/Collect/Components/Collection/Header/DefaultHeader"
import { shallow } from "enzyme"
import React from "react"
import renderer from "react-test-renderer"

describe("artworks", () => {
  it("returns an array of repeating artworks when original array of works is too small to fill header ", () => {
    const artworks = defaultCollectionHeaderArtworks.hits.slice(0, 3)
    const result = getHeaderArtworks(artworks, 1308, false)

    expect(result.length).toBeGreaterThan(artworks.length)
    expect(result).toHaveLength(6)
  })

  it("returns an array of artworks of equal length as the original array received when header can fit all works ", () => {
    const artworks = defaultCollectionHeaderArtworks.hits

    const result = getHeaderArtworks(artworks, 1308, false)
    expect(artworks.length).toEqual(result.length)
    expect(result).toHaveLength(10)
  })
})

describe("default header component", () => {
  let props

  beforeEach(() => {
    props = {
      headerArtworks: defaultCollectionHeaderArtworks,
      defaultHeaderImageHeight: 1000,
    }
  })

  const getWrapper = headerProps => {
    return shallow(<CollectionDefaultHeader {...headerProps} />)
  }

  it("renders a snapshot", () => {
    const component = renderer
      .create(
        <CollectionDefaultHeader
          headerArtworks={props.headerArtworks}
          defaultHeaderImageHeight={props.defaultHeaderImageHeight}
        />
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it("renders images in the header component ", () => {
    const wrapper = getWrapper(props)

    expect(wrapper.find("Image").length).toBe(10)
  })

  it("when viewport size is sm or small the image src link references the small resized url", () => {
    const mockWindow: any = window
    mockWindow.innerWidth = 600
    mockWindow.innerHeight = 600
    const wrapper = getWrapper(props)
    const headerArtwork = wrapper.find("Image").first()

    expect(headerArtwork.props().src).toEqual(
      "https://resized-small.cloudfront.net"
    )
  })

  it("when viewport size is md or larger the image src link references the large resized url", () => {
    const mockWindow: any = window
    mockWindow.innerWidth = 1200
    mockWindow.innerHeight = 1200
    const wrapper = getWrapper(props)
    const headerArtwork = wrapper.find("Image").first()

    expect(headerArtwork.props().src).toEqual(
      "https://resized-large.cloudfront.net"
    )
  })

  it("a header image's anchor tag references the correct artwork slug ", () => {
    const wrapper = getWrapper(props)

    expect(
      wrapper
        .find("a")
        .at(0)
        .props().href
    ).toEqual("/artwork/shepard-fairey-50-shades-of-black-lp-box-set")
  })
})
