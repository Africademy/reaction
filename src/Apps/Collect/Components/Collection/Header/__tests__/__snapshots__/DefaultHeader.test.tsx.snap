// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default header component renders a snapshot 1`] = `
<header>
  <div
    className="sc-bdVaJa DefaultHeader__DefaultHeaderContainer-sc-12cdbk9-0 fVUlqO"
    height={
      Array [
        160,
        160,
        250,
      ]
    }
    width={
      Array [
        "auto",
        1,
      ]
    }
  >
    <div
      className="sc-bwzfXH DefaultHeader__HeaderArtworks-sc-12cdbk9-1 jDVusT"
    >
      <a
        href="/artwork/carrie-mae-weems-untitled-woman-and-daughter-with-children"
      >
        <img
          className="Imageshared__BaseImage-sc-1dymtq1-0 dgNeGC"
          onContextMenu={[Function]}
          src="https://resized-large.cloudfront.net"
        />
      </a>
      <a
        href="/artwork/carrie-mae-weems-untitled-woman-feeding-bird"
      >
        <img
          className="Imageshared__BaseImage-sc-1dymtq1-0 dgNeGC"
          onContextMenu={[Function]}
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=219&height=220&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F6EcgtVvPwlklnB96Xo582Q%2Flarge.jpg"
        />
      </a>
      <a
        href="/artwork/carrie-mae-weems-untitled-man-reading-newspaper-and-kitchen-table-series-book"
      >
        <img
          className="Imageshared__BaseImage-sc-1dymtq1-0 dgNeGC"
          onContextMenu={[Function]}
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=220&height=220&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FgK6jwJ3DvAJwfRN86QVkXg%2Flarge.jpg"
        />
      </a>
      <a
        href="/artwork/carrie-mae-weems-untitled-woman-standing"
      >
        <img
          className="Imageshared__BaseImage-sc-1dymtq1-0 dgNeGC"
          onContextMenu={[Function]}
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=213&height=220&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FclW8TSShyic1OZjPrQ2ZMQ%2Flarge.jpg"
        />
      </a>
      <a
        href="/artwork/carrie-mae-weems-untitled-woman-and-daughter-with-makeup-from-kitchen-table-series-1"
      >
        <img
          className="Imageshared__BaseImage-sc-1dymtq1-0 dgNeGC"
          onContextMenu={[Function]}
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=218&height=220&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F6im4v__40Y7_SkuaZTCVaQ%2Flarge.jpg"
        />
      </a>
    </div>
  </div>
</header>
`;
